### PR TITLE
Added a config setting for "setDomains"

### PR DESCRIPTION
**Pull request by MichaelHeerklotz**
This setting allows to configure the setDomains() call. Related: https://github.com/matomo-org/tag-manager/issues/160

### CODE REVIEW

File: Template/Variable/MatomoConfigurationVariable.php [hunk 1]
```diff
@@ -124,6 +124,13 @@ public function getParameters()
                 $field->description = 'When tracking many subdirectories in separate websites, the cookie path prevents the number of cookies to quickly increase and prevent browser from deleting some of the cookies. This ensures optimal data accuracy and improves performance for your users (fewer cookies are sent with each request).';
                 $field->validators[] = new CharacterLength(0, 500);
             }),
+
+            $this->makeSetting('domains', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = 'Domains';
+                $field->description = 'Set array of hostnames or domains to be treated as local. For wildcard subdomains, you can use: ".example.com" or "*.example.com". You can also specify a path along a domain: "*.example.com/subsite1"';
```
Comment (tsteur) [inline Template/Variable/MatomoConfigurationVariable.php]: Should the description maybe say something like comma separated list? Or how would one actually configure an array? (haven't looked what `setDomains`) supports

Comment (tsteur) [reply Template/Variable/MatomoConfigurationVariable.php]: Just checked `setDomains` expects an array... 
btw be good to have this setting 👍 

I'm not sure if we have a problem or not... haven't tried to implement it and don't know if I remember details 100% correct whether there is a problem when someone wants to define a string to read it from a TagManager Variable or when someone wants to define an array of strings. Not sure if both is actually possible at the same time yet. Not sure you understand.

Comment (MichaelHeerklotz) [reply Template/Variable/MatomoConfigurationVariable.php]: Thank you. The code is now using an array for setDomains().
I have also updated the description a bit, most of it comes from the setDomains() from https://developer.matomo.org/api-reference/tracking-javascript

**Comment by tsteur** [comment]
Cheers 👍  works nicely

**Pull request merged**